features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/ThrottlingPolicies/BlackLists.js

```diff
@@ -0,0 +1,209 @@
+ */
+
+import React, { Component } from 'react'
+
+import { Link } from 'react-router-dom'
+import Grid from 'material-ui/Grid';
+import Paper from 'material-ui/Paper';
+import Typography from 'material-ui/Typography';
+import Table, { TableBody, TableCell, TableHead, TableRow } from 'material-ui/Table';
+import AppBar from 'material-ui/AppBar';
+import Toolbar from 'material-ui/Toolbar';;
+import IconButton from 'material-ui/IconButton';
+import Divider from 'material-ui/Divider';
+import Button from 'material-ui/Button';
+import MenuIcon from 'material-ui-icons/Menu';
+import TextField from 'material-ui/TextField';
+import { withStyles } from 'material-ui/styles';
+import Switch from 'material-ui/Switch';
+
+import API from '../../data/api'
+import Message from '../Shared/Message'
+import Confirm from '../Shared/Confirm'
+import Alert from '../Shared/Alert'
+
+const messages = {
+    success: 'Deleted black list policy successfully',
+    failure: 'Error while deleting black list policy',
+    retrieveError: 'Error while retrieving black list policies'
+};
+
+const styles = theme => ({
+    divider: {
+        marginBottom: 20,
+    },
+    createButton: {
+        textDecoration: 'none',
+        display: 'inline-block',
+        marginLeft: 20,
+        alignSelf: 'flex-start',
+    },
+    titleWrapper: {
+        display: 'flex',
+    }
+});
+
+class BlackLists extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            policies: null,
+            selectedRowKeys: [],
+            open: false,
+            message: ''
+        };
+        this.handlePolicyChange = this.handlePolicyChange.bind(this);
+        this.deleteBlackListPolicy = this.deleteBlackListPolicy.bind(this);
+
+    }
+
+    deleteBlackListPolicy(event) {
+        const api = new API();
+        const id = event.currentTarget.id;
+        const promisedPolicies = api.deleteBlackListPolicy(id);
+        promisedPolicies.then(
+            response => {
+                Alert.info(messages.success);
+                var data = this.state.policies.filter(obj => {
+                    return obj.conditionId !== id;
+                });
+                this.setState({ policies: data });
+            }
+        ).catch(
+            error => {
+                Alert.error(messages.failure);
+                console.error(error);
+            }
+        );
+    }
+
+    componentDidMount() {
+        const api = new API();
+
+        const promised_policies = api.getBlockListPolicies();
+        promised_policies.then(
+            response => {
+                this.setState({ policies: response.obj.list });
+            }
+        ).catch(
+            error => {
+                Alert.error(messages.retrieveError);
+                console.error(error);
+            }
+        );
+    }
+
+    handlePolicyChange(event) {
+        const api = new API();
+        var status = event.target.checked;
+        var id = event.target.id;
```

Comment:
Use object destructuring instead of direct assignment i:e
And please use `let` or `const`
```javascript
cont { id } = event.target;
```

Reply:
I don't see any advantage of object restructuring for a single variable. Then we will have to do this for all const and let